conduit/plugins/importers/syncalgod/syncalgod_importer.go

```diff
@@ -0,0 +1,168 @@
+package syncalgod
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+
+	"github.com/algorand/indexer/conduit"
+	"github.com/algorand/indexer/conduit/plugins"
+	"github.com/algorand/indexer/conduit/plugins/importers"
+	"github.com/algorand/indexer/data"
+
+	"github.com/algorand/go-algorand-sdk/v2/client/v2/algod"
+	sdk "github.com/algorand/go-algorand-sdk/v2/types"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/rpcs"
+)
+
+const importerName = "sync_algod"
+
+type syncModeImporter struct {
+	aclient *algod.Client
+	logger  *logrus.Logger
+	cfg     Config
+	ctx     context.Context
+	cancel  context.CancelFunc
+}
+
+func (sm *syncModeImporter) OnComplete(input data.BlockData) error {
+	_, err := sm.aclient.SetSyncRound(input.Round() + 1).Do(sm.ctx)
+	return err
+}
+
+//go:embed sample.yaml
+var sampleConfig string
+
+var syncAlgodModeImporterMetadata = conduit.Metadata{
+	Name:         importerName,
+	Description:  "Importer for fetching blocks from an algod REST API using synch round and ledger delta algod calls.",
+	Deprecated:   false,
+	SampleConfig: sampleConfig,
+}
+
+// New initializes an algod importer
+func New() importers.Importer {
+	return &syncModeImporter{}
+}
+
+func (sm *syncModeImporter) Metadata() conduit.Metadata {
+	return syncAlgodModeImporterMetadata
+}
+
+// package-wide init function
+func init() {
+	importers.Register(importerName, importers.ImporterConstructorFunc(func() importers.Importer {
+		return &syncModeImporter{}
+	}))
+}
+
+func (sm *syncModeImporter) Init(ctx context.Context, cfg plugins.PluginConfig, logger *logrus.Logger) (*sdk.Genesis, error) {
+	sm.ctx, sm.cancel = context.WithCancel(ctx)
+	sm.logger = logger
+	err := cfg.UnmarshalConfig(&sm.cfg)
+	if err != nil {
+		return nil, fmt.Errorf("connect failure in unmarshalConfig: %v", err)
+	}
+	var client *algod.Client
+	u, err := url.Parse(sm.cfg.NetAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	if u.Scheme != "http" && u.Scheme != "https" {
+		sm.cfg.NetAddr = "http://" + sm.cfg.NetAddr
+		sm.logger.Infof("Algod Importer added http prefix to NetAddr: %s", sm.cfg.NetAddr)
+	}
+	client, err = algod.MakeClient(sm.cfg.NetAddr, sm.cfg.Token)
+	if err != nil {
+		return nil, err
+	}
+	sm.aclient = client
+
+	genesisResponse, err := client.GetGenesis().Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	genesis := sdk.Genesis{}
+
+	err = protocol.DecodeJSON([]byte(genesisResponse), &genesis)
+	if err != nil {
+		return nil, err
+	}
+
+	return &genesis, err
+}
+
+func (sm *syncModeImporter) Config() string {
+	s, _ := yaml.Marshal(sm.cfg)
+	return string(s)
+}
+
+func (sm *syncModeImporter) Close() error {
+	if sm.cancel != nil {
+		sm.cancel()
+	}
+	return nil
```

Comment:
should we call UnsetSyncRound in Close()?

Reply:
If we want to let the node continue catching up once conduit stops then yes.

I think in our case it's best to leave it set so that we can start/stop conduit and the node will still be usable. If we let it catchup freely then when conduit starts again the deltas we need will likely no longer be available.

Because we're making it so much more lightweight to run an algod node for indexer by removing the requirement of it being archival, I think it's not unreasonable to assume that algod nodes service indexer/conduit are dedicated instead of shared across multiple services/use cases.